device/transport/amqp/test/_amqp_twin_client_test.js

```diff
@@ -93,687 +113,448 @@ describe('AmqpTwinClient', function () {
-      var attachSender = sinon.spy(amqpClient, 'attachSenderLink');
-      var attachReceiver = sinon.spy(amqpClient, 'attachReceiverLink');
-      var detachSender = sinon.spy(amqpClient, 'detachSenderLink');
-      var detachReceiver = sinon.spy(amqpClient, 'detachReceiverLink');
-      var newTwinReceiver = new AmqpTwinClient(fakeAuthenticationProvider, amqpClient);
-      newTwinReceiver.on('response', responseFunction);
-      newTwinReceiver.on('post', responseFunction);
-      assert(detachSender.notCalled);
-      assert(detachReceiver.notCalled);
-      newTwinReceiver.removeListener('response', responseFunction);
-      assert(detachSender.notCalled);
-      assert(detachReceiver.notCalled);
-      newTwinReceiver.removeListener('post', responseFunction);
-      assert(detachSender.calledOnce);
-      assert(detachReceiver.calledOnce);
-    });
-
-    /* Tests_SRS_NODE_DEVICE_AMQP_TWIN_06_015: [If there is a listener for the `response` event, a `response` event shall be emitted for each response received for requests initiated by SendTwinRequest.] */
-    /* Tests_SRS_NODE_DEVICE_AMQP_TWIN_06_026: [The `status` value is acquired from the amqp message status message annotation.] */
-    /* Tests_SRS_NODE_DEVICE_AMQP_TWIN_06_016: [When a `response` event is emitted, the parameter shall be an object which contains `status`, `requestId` and `body` members.] */
-    /* Tests_SRS_NODE_DEVICE_AMQP_TWIN_06_017: [The `requestId` value is acquired from the amqp message correlationId property in the response amqp message.] */
-    /* Tests_SRS_NODE_DEVICE_AMQP_TWIN_06_018: [The `body` parameter of the `response` event shall be the data of the received amqp message.] */
-    it('Emit a response message for a sendTwinRequest.', function(responseDone) {
-      var amqpClient = new AmqpProvider();
-      var messageData = 'a string for the body';
-      var newTwinReceiver = new AmqpTwinClient(fakeAuthenticationProvider, amqpClient);
-      newTwinReceiver.on('subscribed', function(event) {
-        if (event.eventName === 'response') {
-          process.nextTick(function () {
-            newTwinReceiver.sendTwinRequest('GET', '/abc', { '$rid' : 10 }, messageData, function() {});
-            //
-            // Form a common message for the receiver link to emit.
-            //
-            var newMessage = new Message();
-            newMessage.correlationId = '10';
-            newMessage.data = messageData;
-            newMessage.transportObj = {};
-            newMessage.transportObj.messageAnnotations = {};
-            newMessage.transportObj.messageAnnotations['status'] = 201;
-            amqpClient.fakeReceiverLink.emit('message', newMessage);
-          });
-        }
+    it('configures link options for the sender and receiver links', function() {
+      methodUnderTest();
+      assert.equal(fakeAmqpClient.attachSenderLink.firstCall.args[1].attach.sndSettleMode, 1, ' sender send settle mode not set appropriately' );
```

Comment:
>fakeAmqpClient.attachSenderLink.firstCall.args[1].attach [](start = 19, length = 56)

nit: Looks like a perfect place to use a temporary to hold fakeAmqpClient.attachSenderLink.firstCall.args[1].attach and fakeAmqpClient.attachReceiver.firstCall.args[1].attach :)